client/blocks/site/index.jsx

```diff
@@ -68,6 +73,33 @@ class Site extends React.Component {
 		this.props.onMouseLeave( event, this.props.site.ID );
 	};
 
+	onViewSiteClick = event => {
+		const { isPreviewable, siteSuffix } = this.props;
+
+		if ( ! isPreviewable ) {
+			this.trackMenuItemClick( 'view_site_unpreviewable' );
+			this.props.recordGoogleEvent( 'Sidebar', 'Clicked View Site | Unpreviewable' );
+			return;
+		}
+
+		if ( event.altKey || event.ctrlKey || event.metaKey || event.shiftKey ) {
+			this.trackMenuItemClick( 'view_site_modifier' );
+			this.props.recordGoogleEvent( 'Sidebar', 'Clicked View Site | Modifier Key' );
+			return;
+		}
+
+		event.preventDefault();
```

Comment:
The `preventDefault` call is not needed. `button` doesn't have any default action to be prevented. This line is only useful for, e.g., links that have both onclick handler and a href attribute:
```jsx
<a href="/view/site.slug" onClick={ handleClick }>View</a>
```
Then the `handleClick` handler can prevent the `href` navigation from occuring.